src/algebra/module.lean

```diff
@@ -22,25 +22,26 @@ infixr ` • `:73 := has_scalar.smul
   connected by a "scalar multiplication" operation `r • x : β`
   (where `r : α` and `x : β`) with some natural associativity and
   distributivity axioms similar to those on a ring. -/
-class semimodule (α : out_param $ Type u) (β : Type v) [out_param $ semiring α]
+class semimodule (α : Type u) (β : Type v) [semiring α]
   [add_comm_monoid β] extends has_scalar α β :=
-(smul_add : ∀r (x y : β), r • (x + y) = r • x + r • y)
-(add_smul : ∀r s (x : β), (r + s) • x = r • x + s • x)
-(mul_smul : ∀r s (x : β), (r * s) • x = r • s • x)
+(smul_add : ∀(r : α) (x y : β), r • (x + y) = r • x + r • y)
+(add_smul : ∀(r s : α) (x : β), (r + s) • x = r • x + s • x)
+(mul_smul : ∀(r s : α) (x : β), (r * s) • x = r • s • x)
 (one_smul : ∀x : β, (1 : α) • x = x)
 (zero_smul : ∀x : β, (0 : α) • x = 0)
-(smul_zero {} : ∀r, r • (0 : β) = 0)
+(smul_zero {} : ∀(r : α), r • (0 : β) = 0)
 
 section semimodule
-variables {R:semiring α} [add_comm_monoid β] [semimodule α β] (r s : α) (x y : β)
+variables [R:semiring α] [add_comm_monoid β] [semimodule α β] (r s : α) (x y : β)
 include R
 
 theorem smul_add : r • (x + y) = r • x + r • y := semimodule.smul_add r x y
 theorem add_smul : (r + s) • x = r • x + s • x := semimodule.add_smul r s x
 theorem mul_smul : (r * s) • x = r • s • x := semimodule.mul_smul r s x
-@[simp] theorem one_smul : (1 : α) • x = x := semimodule.one_smul x
-@[simp] theorem zero_smul : (0 : α) • x = 0 := semimodule.zero_smul x
 @[simp] theorem smul_zero : r • (0 : β) = 0 := semimodule.smul_zero r
+variables (α)
+@[simp] theorem one_smul : (1 : α) • x = x := semimodule.one_smul α x
+@[simp] theorem zero_smul : (0 : α) • x = 0 := semimodule.zero_smul α x
```

Comment:
Why is alpha explicit here? It looks like it messed up the `pi_instance` for semimodules in the other file.

Reply:
Also `pi_instance` is still messed up after I made it implicit, so it is not the cause

Reply:
`pi_instance` fails for `smul` because `pi_instance` tries to apply `has_scalar.smul` but it fails because the instance cannot be synthesized